src/Libraries/RevitNodes/Elements/Parameter.cs

```diff
@@ -40,5 +205,190 @@ public override string ToString()
+            Autodesk.Revit.DB.BuiltInParameterGroup parameterGroup = Autodesk.Revit.DB.BuiltInParameterGroup.PG_DATA;
+            if (!System.Enum.TryParse<Autodesk.Revit.DB.BuiltInParameterGroup>(group, out parameterGroup))
+                throw new System.Exception(Properties.Resources.ParameterTypeNotFound);
+
+            // get document and open transaction
+            Document document = Application.Document.Current.InternalDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            try
+            {
+                // buffer the current shared parameter file name and apply a new empty parameter file instead
+                string sharedParameterFile = document.Application.SharedParametersFilename;
+                string tempSharedParameterFile = System.IO.Path.GetTempFileName() + ".txt";
+                using (System.IO.File.Create(tempSharedParameterFile)) { }
+                document.Application.SharedParametersFilename = tempSharedParameterFile;
+
+                // Apply selected parameter categories
+                CategorySet categories = ToCategorySet(categoryList);
+
+                // create a new shared parameter, since the file is empty everything has to be created from scratch
+                ExternalDefinition def = document.Application.OpenSharedParameterFile().Groups.Create(groupName).Definitions.Create(new ExternalDefinitionCreationOptions(parameterName, parameterType)) as ExternalDefinition;
+
+                // Create an instance or type binding
+                Binding bin = (instance) ? (Binding)document.Application.Create.NewInstanceBinding(categories) : (Binding)document.Application.Create.NewTypeBinding(categories);
+
+                // Apply parameter bindings
+                document.ParameterBindings.Insert(def, bin, parameterGroup);
+
+                // apply old shared parameter file
+                document.Application.SharedParametersFilename = sharedParameterFile;
+
+                // delete temp shared parameter file
+                System.IO.File.Delete(tempSharedParameterFile);
+
+
+            }
+            catch (System.Exception e)
+            {
+                TransactionManager.Instance.ForceCloseTransaction();
+                throw e;
+            }
+
+            TransactionManager.Instance.TransactionTaskDone();
+        }
+
+        #endregion
+
+        /// <summary>
+        /// Transform List of categories to CategorySet
+        /// </summary>
+        /// <param name="categoryList"></param>
+        /// <returns></returns>
+        internal static CategorySet ToCategorySet(System.Collections.Generic.IEnumerable<Category> categoryList)
+        {
+            Document document = Application.Document.Current.InternalDocument;
+
+            // Apply selected parameter categories
+            CategorySet categories = document.Application.Create.NewCategorySet();
+
+            if (categoryList == null)
```

Comment:
if you can create that allCategory token you would handle that case before calling this method I guess